design/aws-ecs-scheduled-ecs-task-construct.md

```diff
@@ -0,0 +1,131 @@
+
+## Code changes
+
+Given the above, we should make the following changes to support scheduled tasks on ECS:
+1. Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct
+
+# Part 1: Create `ScheduledEc2TaskProps` interface  and `ScheduledEc2Task` construct  
+
+The `ScheduledEc2TaskProps` interface will contain properties to construct the Ec2TaskDefinition, Ec2EventRuleTarget and EventRule:
+
+```ts
+export interface ScheduledEc2TaskProps {
+  /**
+   * The cluster where your service will be deployed.
+   */
+  readonly cluster: ICluster;
+
+  /**
+   * The CMD value to pass to the container.
+   *
+   * If you provide a shell command as a single string, you have to quote command-line arguments.
+   */
+  readonly command?: string[];
+
+  /**
+   * The minimum number of CPU units to reserve for the container.
+   */
+  readonly cpu?: number;
+
+  /**
+   * Number of desired copies of running tasks.
+   *
+   * @default 1
+   */
+  readonly desiredTaskCount?: number;
+
+  /**
+   * The environment variables to pass to the container.
+   */
+  readonly environment?: { [key: string]: string };
+
+  /**
+   * The image to start.
+   */
+  readonly image: ContainerImage;
+
+  /**
+   * The hard limit (in MiB) of memory to present to the container.
+   *
+   * If your container attempts to exceed the allocated memory, the container
+   * is terminated.
+   *
+   * At least one of memoryLimitMiB and memoryReservationMiB is required for non-Fargate services.
+   *
+   * @default 256
+   */
+  readonly memoryLimitMiB?: number;
+
+  /**
+   * The soft limit (in MiB) of memory to reserve for the container.
+   *
+   * When system memory is under contention, Docker attempts to keep the
+   * container memory within the limit. If the container requires more memory,
+   * it can consume up to the value specified by the Memory property or all of
+   * the available memory on the container instance—whichever comes first.
+   *
+   * At least one of memoryLimitMiB and memoryReservationMiB is required for non-Fargate services.
+   */
+  readonly memoryReservationMiB?: number;
+
+  /**
+   * The schedule or rate (frequency) that determines when CloudWatch Events
+   * runs the rule. For more information, see Schedule Expression Syntax for
+   * Rules in the Amazon CloudWatch User Guide.
+   *
+   * @see http://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html
+   *
+   * You must specify this property.
+   */
+  readonly scheduleExpression?: string;
+}
+```
+
+The `ScheduledEc2Task` construct will use the following existing constructs:
+
+* Ec2TaskDefinition - To create a Task Definition for the container to start
+* Ec2EventRuleTarget - The target of the aws event
```

Comment:
Should be an EcsEventRuleTarget? :)

Reply:
The existing L2 construct is actually called Ec2EventRuleTarget: https://github.com/awslabs/aws-cdk/blob/master/packages/%40aws-cdk/aws-ecs/lib/ec2/ec2-event-rule-target.ts

Reply:
Oh weird that it's not in @aws-cdk/aws-events-targets